src/meta/src/rpc/ddl_controller.rs

```diff
@@ -1620,16 +1621,11 @@ impl DdlController {
         let cluster_info = self.metadata_manager.get_streaming_cluster_info().await?;
 
         let parallelism = self.resolve_stream_parallelism(specified_parallelism, &cluster_info)?;
-
-        // TODO(var-vnode): use vnode count from config
-        const MAX_PARALLELISM: NonZeroUsize = NonZeroUsize::new(VirtualNode::COUNT).unwrap();
-
-        let parallelism_limited = parallelism > MAX_PARALLELISM;
+        let parallelism_limited = parallelism > max_parallelism;
         if parallelism_limited {
-            tracing::warn!("Too many parallelism, use {} instead", MAX_PARALLELISM);
+            tracing::warn!("Too many parallelism, use {} instead", max_parallelism);
```

Comment:
```suggestion
            tracing::warn!("Parallelism cannot be greater than vnode count {}. Use {} instead", max_parallelism, max_parallelism);
```

Not sure whether we expose the term "vnode count" to users. If not, please use other proper term.

Reply:
Actually, I tend to return error here, instead of silently warning.

Reply:
Agree. We may address them together in the user-facing PR: https://github.com/risingwavelabs/risingwave/pull/18515/files#diff-776f2681803414a56f539f2c98cc63a749bad4d8531d3d3191fc5df9c7efc004R61